dxp/demystifying_aui_input_tags.md

```diff
@@ -0,0 +1,727 @@
+# AUI Input
+
+## Table of Contents
+
+<div><a class="f5" href="#user-content-text">Text</a></div>
+<div><a class="f5" href="#user-content-textarea">Textarea</a></div>
+<div><a class="f5" href="#user-content-range">Range</a></div>
+
+The `<aui:input />` taglib is a custom tag used in `jsp` files to generate HTML form elements that conform to Liferay Portal's markup patterns, provide language translations, and other convenience methods. We should use this when adding form elements inside `jsp` files.
+
+The only required attribute is `name`, without it the `jsp` build will fail. The example below shows the minimum amount of code required for the taglib to output markup.
+
+**⚠️ Counterexample:**
+
+```jsp
+<aui:input name="firstName" />
+```
+
+If a `type` attribute is not declared, the taglib will, most likely, default to `text`, but depends on the combination of attributes that are declared in the `aui:input`. This will be explained in detail later in this document. We always recommend specifying a `type` when creating a standard HTML form element to reduce confusion since this taglib is a catch all for Liferay form elements.
```

Comment:
It also depends on the `bean`. If there's an associated bean which definition can be found in `model-hints.xml` then the type of input can be inferred from the name and the definition.

At least that's what I think is going on, but we'll need @izaera's help to clarify.

Reply:
- "depends" → "it depends"
- "catch all" → "catch-all"

Reply:
One day I'll be able to document this one...

![pray](https://media.giphy.com/media/3oz8xALrfMNLmcNrVe/giphy.gif)